EIPS/eip-draft_deposit_contract_snapshot_interface.md

```diff
@@ -0,0 +1,282 @@
+    right: MerkleTree
+    def get_root(self) -> Hash32:
+        return sha256(self.left.get_root() + self.right.get_root())
+    def is_full(self) -> bool:
+        return self.right.is_full()
+    def push_leaf(self, leaf: Hash32, level: uint) -> MerkleTree:
+        if not(self.left.is_full()):
+            self.left = self.left.push_leaf(leaf, level - 1)
+        else:
+            self.right = self.right.push_leaf(leaf, level - 1)
+        return self
+    def finalize(self, deposits_to_finalize: uint, level: uint) -> MerkleTree:
+        deposits = 2**level
+        if deposits <= deposits_to_finalize:
+            return Finalized(deposits, self.get_root())
+        self.left = self.left.finalize(deposits_to_finalize, level - 1)
+        if deposits_to_finalize > deposits / 2:
+            remaining = deposits_to_finalize - deposits / 2
+            self.right = self.right.finalize(remaining, level - 1)
+        return self
+    def get_finalized(self, result: List[Hash32]) -> uint:
+        return self.left.get_finalized(result) + self.right.get_finalized(result)
+
+@dataclass
+class Zero(MerkleTree):
+    n: uint64
+    def get_root(self) -> Hash32:
+        return zerohashes[self.n]
+    def is_full(self) -> bool:
+        return False
+    def push_leaf(self, leaf: Hash32, level: uint) -> MerkleTree:
+        return MerkleTree.create([leaf], level)
+    def get_finalized(self, result: List[Hash32]) -> uint:
+        return 0
+```
+
+## Test Cases
+
+Test cases are included in [test_cases.yaml](../assets/eip-draft_deposit_contract_snapshot_interface/test_cases.yaml). Each case is structured as follows:
+
+```python
+class DepositTestCase:
+    deposit_data: DepositData     # These are all the inputs to the deposit contract's deposit() function
+    deposit_data_root: Hash32     # The tree hash root of this deposit (calculated for convenience)
+    eth1_data: Eth1Data           # An Eth1Data object that can be used to finalize the tree after pushing this deposit
+    snapshot: DepositTreeSnapshot # The resulting DepositTreeSnapshot object if the tree were finalized after this deposit
+```
+
+This EIP also includes other files for testing:
+* [deposit_snapshot.py](../assets/eip-draft_deposit_contract_snapshot_interface/deposit_snapshot.py) contains the same code as the [Reference Implementation](#reference-implementation)
+* [eipDRAFT.py](../assets/eip-draft_deposit_contract_snapshot_interface/eipDRAFT.py) contains boilerplate declarations
+* [test_deposit_snapshot.py](../assets/eip-draft_deposit_contract_snapshot_interface/test_deposit_snapshot.py) includes code for running test cases against the reference implementation
+
+If these files are download to the same directory, the test cases can be run by executing `pytest` in that directory.
+
+#### Existing Implementations
+* [lighthouse](https://github.com/sigp/lighthouse/pull/2915)
+* [nimbus](https://github.com/status-im/nimbus-eth2/commit/372c9b798c005102c7777a16aaa7c822267330fa)
+
```

Comment:
```suggestion
```
This sort of data shouldn't be in the EIP, it should be in some other material.  An easy place to track this sort of thing is the discussions-to link, but I have seen good success by creating a hackmd page or similar that is passed around (which links both to the EIP itself and external clients like this).

Reply:
Done! I'll put the existing implementations in the ETH magicians discussion thread :)